go/vt/vtctl/reparentutil/replication.go

```diff
@@ -142,6 +142,10 @@ func FindValidEmergencyReparentCandidates(
 // ReplicaWasRunning returns true if a StopReplicationStatus indicates that the
 // replica had running replication threads before being stopped.
 func ReplicaWasRunning(stopStatus *replicationdatapb.StopReplicationStatus) bool {
+	if stopStatus.Before == nil {
```

Comment:
I'm not sure false is the right thing to return here. If before is nil we can't know if it was running or not. Not getting `Before` at all, IMO, should be a panic case.

Reply:
I think we shouldn't have code in the critical path of an ERS that can panic, because we'd either need to add `recover` defers and figure out how to recover from ... any point where the process can fail, or risk leaving the cluster in a worse state than when we started.

How about:

```go
func ReplicaWasRunning(stopStatus *replicationdatapb.StopReplicationStatus) (bool, error) {
	if stopStatus.Before == nil {
		return false, errors.New("...")
	}

	return stopStatus.Before.IoThreadRunning || stopStatus.Before.SqlThreadRunning, nil
}
```

Reply:
Better error than panic :) This lgtm.

Reply:
This looks better